### PR TITLE
8255918: XMLStreamFilterImpl constructor consumes XMLStreamException

### PR DESCRIPTION
**Pull request by MikeEdgar**
Allow `XMLStreamException` to be thrown to the application when a filtered `XMLStreamReader` encounters an `XMLStreamException` advancing the wrapped `XMLStreamReader`.

Note, this PR includes a method signature change (constructor of `com.sun.org.apache.xerces.internal.impl.XMLStreamFilterImpl`).

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8255918](https://bugs.openjdk.java.net/browse/JDK-8255918): XMLStreamFilterImpl constructor consumes XMLStreamException

### Reviewers
 * [Joe Wang](https://openjdk.java.net/census#joehw) (@JoeWang-Java - **Reviewer**) ⚠️ Review applies to a416b2768f272c29019b0b07d5c1862fbda5cd07

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/1209/head:pull/1209`
`$ git checkout pull/1209`

### CODE REVIEW

**Comment by MikeEdgar** [comment]
Test has been updated for `jtreg`.

File: test/jdk/javax/xml/jaxp/parsers/8255918/XMLStreamReaderFilterTest.java
Comment (JoeWang-Java) [inline test/jdk/javax/xml/jaxp/parsers/8255918/XMLStreamReaderFilterTest.java:22]: Add package stream.XMLStreamFilterTest;

File: test/jdk/javax/xml/jaxp/parsers/8255918/XMLStreamReaderFilterTest.java
Comment (JoeWang-Java) [inline test/jdk/javax/xml/jaxp/parsers/8255918/XMLStreamReaderFilterTest.java:75]: It would be good/sufficient to verify XMLStreamException with assertThrows. Checking the message details can be fragile.

File: test/jdk/javax/xml/jaxp/parsers/8255918/XMLStreamReaderFilterTest.java
Comment (JoeWang-Java) [inline test/jdk/javax/xml/jaxp/parsers/8255918/XMLStreamReaderFilterTest.java:58]: Tags are added in the class description. Change this to a TestCase javadoc or note.

**Review by JoeWang-Java — COMMENTED** [review]
I left comments the other day but forgot to submit. I've adjusted them based on your new changes.

File: test/jdk/javax/xml/jaxp/parsers/8255918/XMLStreamReaderFilterTest.java
Comment (JoeWang-Java) [reply test/jdk/javax/xml/jaxp/parsers/8255918/XMLStreamReaderFilterTest.java:38]: This comment was shown under "static final String XMLSOURCE1" in the email, though in "Files changed" pane it still is right below the class description "JDK-8255918".

The comment meant to be class description, replacing "JDK-8255918".

**Comment by MikeEdgar** [comment]
@JoeWang-Java - I believe I have addressed all of your comments. The branch has also been re-based on the `master` branch. Please let me know if any additional changes are needed.

File: test/jaxp/javax/xml/jaxp/unittest/stream/XMLStreamFilterTest/XMLStreamReaderFilterTest.java
Comment (JoeWang-Java) [inline test/jaxp/javax/xml/jaxp/unittest/stream/XMLStreamFilterTest/XMLStreamReaderFilterTest.java:60]: This test verifies an assertion made by the javadoc for the createFilteredReader method. It would be good to made the first sentence a summary of the test. Something like:

Verifies that XMLStreamException is thrown as specified by the {@code XMLInputFactory::createFilteredReader} method when an error is encountered. This test illustrates the scenario by creating a reader with a filter that requires the original reader to advance past the invalid element in the underlying XML.

File: test/jaxp/javax/xml/jaxp/unittest/stream/XMLStreamFilterTest/XMLStreamReaderFilterTest.java
Comment (JoeWang-Java) [inline test/jaxp/javax/xml/jaxp/unittest/stream/XMLStreamFilterTest/XMLStreamReaderFilterTest.java:64]: Method name is a bit long. Since the class is dedicated to testing StreamFilter, its test cases may just indicate their particular cases. I would think testCreateFilteredReader is sufficient.

File: test/jaxp/javax/xml/jaxp/unittest/stream/XMLStreamFilterTest/XMLStreamReaderFilterTest.java
Comment (JoeWang-Java) [inline test/jaxp/javax/xml/jaxp/unittest/stream/XMLStreamFilterTest/XMLStreamReaderFilterTest.java:65]: Keeping the length to under 100 characters is still a good practice even with wide screens, 80 - 100 would be good as it's easier to read with a side-by-side view.

**Review by JoeWang-Java — APPROVED** [review]
Looks good!  Some minor comments below.

**Comment by MikeEdgar** [comment]
@JoeWang-Java - thanks for the additional feedback - all good. Do you suggest I make them now in this PR? Just confirming before pushing the change and dismissing your approval.

**Comment by JoeWang-Java** [comment]
Yes, please make the changes. Approval will be quick :-) (since this is the last day for low priority changes)

**Comment by MikeEdgar** [comment]
Done - please let me know if anything else is needed.

**Pull request closed without merging**
